gobblin-utility/src/main/java/gobblin/util/AutoCloseableLock.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.util;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.Lock;
+
+import lombok.AllArgsConstructor;
+
+
+/**
+ * An auto-closeable {@link Lock} that can be used with try-with-resources.
+ */
+@AllArgsConstructor
+public class AutoCloseableLock implements Lock, AutoCloseable {
```

Comment:
I would prefer if the lock automatically locked on construction, and to make everything private but `close()`. That way, the usage must always be:
```java
try (new AutoCloseableLock(lock)) {
...
}
```